doc/source/extended_documentation/calibration/rainforests_calibration/rainforests_calibration.rst

```diff
@@ -0,0 +1,173 @@
+so-called super-ensemble, which we subsequently sample to produce the calibrated forecast
+ensemble output.
+
+Another point of difference between RainForests and ECPoint is in the choice of error value
+underlying the error distribution. For RainForests we have chosen to use additive error in
+place of a multiplicative error (forecast error ratio in ECPoint) to allow us to calibrate
+input forecast values of zero-rainfall.
+
+===========================
+GBDT vs manually trained DT
+===========================
+
+The choice of using GBDT models in place of the manually trained DT of ECPoint comes with
+some advantages, but at the expense of some trade-offs:
+
+**Advantages:**
+
+* GBDT is an ensemble of many trees rather than a single tree. This means outputs are
+  near-continuous relative to the inputs.
+* Trees are built algorithmically, not manually, with each branch split automatically
+  chosen to be optimal relative to some loss function. In principle this gives better
+  accuracy, and makes it easier to retrain on new data.
+
+**trade-offs:**
+
+* By using an ensemble of trees, the intuitive connection between weather type and feature
+  variables is lost.
+* Using a series of decision tree ensembles in place of a single decision tree increases the
+  computational demand significantly.
+
+****************************
+Implementation details
+****************************
+
+===========================
+Model training
+===========================
+
+..
+    TODO: Add more specific details when model training Plugin is incorporated into IMPROVER.
+
+The model training process is relatively simple and involves collating a series of
+forecast-observation pairs with the associated feature variables into a single pandas
+dataframe.
+
+Using this dataframe, a series of lightGBM binary-classification models are trained against
+the truth values for exceedances of the forecast error relative to a series of error-threshold
+values, resulting in one model per error threshold.
+
+As the calibration method works on a per realization basis, the errors used in training
+must be representative of the underlying systematic model biases for a single ensemble
+member rather than those of the ensemble as a whole. This requires the use of a single
+representative ensemble forecast member in training. It also requires the underlying
+weather type associated with the forecast and observation in the forecast-observation
+pair be as consistent as possible.
+
+For the forecast values used in training, we use the control at the shortest available
+lead-times that distinctly represent of the underlying weather types. Specifically, we
+take the earliest 24-hour period for daily accumulations, and all lead-times over the
+earliest 24-hour period for sub-daily accumulations to capture the full cycle of diurnal
+variations.
+
+Currently model training is done offline, using a minimum 12-month period to capture the
+full seasonal cycle.
```

Comment:
Could add a comment that an advantage of the RainForests approach is that with suitable feature variable choice, calibration can be implicitly and automatically localised in space and time where the calibration process determines there is skill to do so, without having to manually discretise the input data in space or time.  For example, use of solar energy input as a feature variable implicitly provides some  degree of seasonal variation in calibration.  Again depends upon what's considered appropriate here vs in external documentation.

Reply:
Good suggestion, however as this is also applicable to ECPoint I will add this to the opening section: **Sub-grid variability as a means of calibration**.